python/ray/train/tests/test_torch_trainer.py

```diff
@@ -114,6 +125,40 @@ def __call__(self, x):
     assert predictions.count() == 3
 
 
+def test_torch_e2e_dir(ray_start_4_cpus, tmpdir):
+    def train_func():
+        model = torch.nn.Linear(1, 1)
+        torch.save(model, os.path.join(tmpdir, "model"))
+        session.report({}, checkpoint=TorchCheckpoint.from_directory(tmpdir))
+
+    scaling_config = ScalingConfig(num_workers=2)
+    trainer = TorchTrainer(
+        train_loop_per_worker=train_func,
+        scaling_config=scaling_config,
+        preprocessor=DummyPreprocessor(),
+    )
+    result = trainer.fit()
+    isinstance(result.checkpoint.get_preprocessor(), DummyPreprocessor)
+
+    class TorchScorer:
+        def __init__(self):
+            with result.checkpoint.as_directory() as checkpoint_path:
+                model = torch.load(os.path.join(checkpoint_path, "model"))
+            preprocessor = result.checkpoint.get_preprocessor()
+            self.pred = TorchPredictor.from_checkpoint(
+                TorchCheckpoint.from_model(model, preprocessor=preprocessor)
+            )
+
+        def __call__(self, x):
+            return self.pred.predict(x, dtype=torch.float)
+
+    predict_dataset = ray.data.range(3)
+    predictions = predict_dataset.map_batches(
+        TorchScorer, batch_format="pandas", compute="actors"
```

Comment:
does TorchPredictor not work directly?

Reply:
I didn't modify the test aside from just making it work - let me see if I can use the predictor here directly